bank/bank.go

```diff
@@ -1,55 +1,83 @@
 package bank
 
 import (
+	"errors"
+	"regexp"
 	"strconv"
 )
 
-type response struct {
-	bankName string
-	isValid  bool
+var bankCode = map[string]string{
+	"636214": "ayandeh",
+	"627412": "eghtesad-novin",
+	"627381": "ansar",
+	"505785": "iran-zamin",
+	"622106": "parsian",
+	"627884": "parsian",
+	"639194": "parsian",
+	"502229": "pasargad",
+	"502249": "pasargad",
+	"639347": "pasargad",
+	"585983": "tejarat",
+	"627353": "tejarat",
+	"502908": "tosee-taavon",
+	"207177": "export-development",
+	"627648": "export-development",
+	"636949": "hekmat-iranian",
+	"505809": "middle-east",
+	"585947": "middle-east",
+	"502938": "dey",
+	"589463": "refah",
+	"621986": "saman",
+	"589210": "sepah",
+	"604932": "sepah",
+	"639607": "sarmayeh",
+	"639346": "sina",
+	"502806": "shahr",
+	"504706": "shahr",
+	"603769": "saderat",
+	"903769": "saderat",
+	"627961": "industry-and-mine",
+	"504172": "resalat",
+	"606373": "mehr-iran",
+	"639599": "ghavamin",
+	"502910": "karafarin",
+	"627488": "karafarin",
+	"603770": "keshavarzi",
+	"639217": "keshavarzi",
+	"505416": "tourism",
+	"636795": "central-bank",
+	"628023": "maskan",
+	"610433": "mellat",
+	"991975": "mellat",
+	"170019": "melli",
+	"603799": "melli",
+	"639370": "mehr-eqtesad",
+	"627760": "post-bank",
+	"628157": "tosseh",
+	"606256": "melal",
+	"505801": "kosar",
+	"507677": "noor",
 }
 
-type Card struct {
-	Number string
-}
-
-func (c *Card) CardInfo() response {
```

Comment:
May I ask you please explain this? why we have remove pointer? when you use pointers your application will be faster, because you’ll avoid copying data around all the time. and our data is a constant

Reply:
your implement with method & I implementation with function. your implementation with method is not faster than function! I just use function because it's shorter & easiest to use.